source/patterns/@aws-solutions-constructs/core/lib/websocket-api-helper.ts

```diff
@@ -0,0 +1,206 @@
+import * as apigwv2 from "aws-cdk-lib/aws-apigatewayv2";
+import { WebSocketAwsIntegration } from "aws-cdk-lib/aws-apigatewayv2-integrations";
+import * as iam from "aws-cdk-lib/aws-iam";
+import * as logs from "aws-cdk-lib/aws-logs";
+import * as sqs from "aws-cdk-lib/aws-sqs";
+import { Construct } from "constructs";
+import { buildLogGroup } from "./cloudwatch-log-group-helper";
+import { addCfnSuppressRules, consolidateProps } from "./utils";
+import { DEFAULT_ROUTE_QUEUE_VTL_CONFIG } from "./websocket-api-defaults";
+
+export interface BuildWebSocketQueueApiResponse {
+  readonly webSocketApi: apigwv2.WebSocketApi;
+  readonly webSocketStage: apigwv2.WebSocketStage;
+  readonly apiGatewayRole: iam.Role;
+  readonly apiGatewayLogGroup: logs.LogGroup;
+}
+
+export interface BuildWebSocketApiProps {
+  /**
+   * Existing instance of ApiGateway v2 WebSocket
+   */
+  readonly existingWebSocketApi?: apigwv2.WebSocketApi;
+
+  /**
+   * User provided properties of Apigateway v2 WebSocket
+   */
+  readonly webSocketApiProps?: apigwv2.WebSocketApiProps;
+}
+
+export interface BuildWebSocketQueueApiRequest {
+  readonly queue: sqs.IQueue;
+  // TODO: Should we create an interface representing { [contentType: string]: string; }? It would be in code documentation
+  readonly defaultRouteRequestTemplate?: { [contentType: string]: string };
+  readonly createDefaultRoute?: boolean;
+  readonly webSocketApiProps?: apigwv2.WebSocketApiProps;
+  readonly existingWebSocketApi?: apigwv2.WebSocketApi;
+  readonly logGroupProps?: logs.LogGroupProps;
+}
+/**
+ * Builds an AWS API Gateway WebSocket API integrated with an Amazon SQS queue.
+ *
+ * @param scope The construct scope where the resources will be created.
+ * @param id The unique ID for the resources.
+ * @param props The configuration properties for the WebSocket API and SQS queue integration.
+ * @returns
+ */
+export function buildWebSocketQueueApi(
+  scope: Construct,
+  id: string,
+  props: BuildWebSocketQueueApiRequest,
+): BuildWebSocketQueueApiResponse {
+  // Setup the API Gateway role
+  const apiGatewayRole = new iam.Role(scope, "LambdaRestApiCloudWatchRole", {
+    assumedBy: new iam.ServicePrincipal("apigateway.amazonaws.com"),
+  });
+  props.queue.grantSendMessages(apiGatewayRole);
+  const webSocketApi = buildApiGatewayV2WebSocket(scope, id, {
+    webSocketApiProps: consolidateProps(
+      buildWebSocketApiProps(
+        apiGatewayRole,
+        props.queue,
+        props.createDefaultRoute,
+        props.defaultRouteRequestTemplate
+      ),
+      props.webSocketApiProps
+    ),
+    existingWebSocketApi: props.existingWebSocketApi,
+  });
+
+  const webSocketStage = new apigwv2.WebSocketStage(scope, "Stage", {
+    stageName: "prod",
+    webSocketApi,
+    autoDeploy: true,
+  });
+
+  const apiGatewayLogGroup = buildLogGroup(scope, "LogGroup", props.logGroupProps);
+  apiGatewayLogGroup.grant(
+    apiGatewayRole,
+    ...[
```

Comment:
Wouldn't this work exactly the same if you removed the ...[ and ]? If the list is fixed and present, why go through the effort to turn it into an array then back into a list? If this is correct, the spread operator just adds confusion (which seems to be a consistent trait of the spread operator)

Reply:
removed the spread operator and array notation.